Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -4417,28 +4420,49 @@ static RefPtr<CSSValue> consumeGeneratedImage(CSSParserTokenRange& range, const
     return result;
 }
 
+static auto consumeImageSetOptionalArguments(CSSParserTokenRange& range)
```

Comment:
Thinking about this approach a bit more, I don't really like it as much.

I think for the parsing I'm going to try to adopt the Meta Consumer objects that are elsewhere in this file. Something like
```C++
struct ImageSetOptionalArgumentConsumer {
    using FunctionToken = ImageSetOptionalTypeArgumentConsumer;
    using DimensionToken = ResolutionCSSPrimitive(...)Consumer;
};
```